modules/imgcodecs/test/test_animation.cpp

```diff
@@ -305,6 +360,51 @@ TEST(Imgcodecs_APNG, imwriteanimation_rgba)
     EXPECT_EQ(0, remove(output.c_str()));
 }
 
+TEST(Imgcodecs_APNG, imwriteanimation_rgba16u)
+{
+    Animation s_animation, l_animation;
+    EXPECT_TRUE(fillFrames(s_animation, true));
+
+    for (size_t i = 0; i < s_animation.frames.size(); i++)
+    {
+        s_animation.frames[i].convertTo(s_animation.frames[i], CV_16U, 255);
+    }
+    // Create a temporary output filename for saving the animation.
+    string output = cv::tempfile(".png");
+
+    // Write the animation to a .png file and verify success.
+    EXPECT_TRUE(imwriteanimation(output, s_animation));
+
+    // Read the animation back and compare with the original.
+    EXPECT_TRUE(imreadanimation(output, l_animation));
+
+    size_t expected_frame_count = s_animation.frames.size() - 2;
+
+    // Verify that the number of frames matches the expected count.
+    EXPECT_EQ(expected_frame_count, imcount(output));
+    EXPECT_EQ(expected_frame_count, l_animation.frames.size());
+
+    std::vector<uchar> buf;
+    readFileBytes(output, buf);
+    vector<Mat> apng_frames;
+
+    EXPECT_TRUE(imdecodemulti(buf, IMREAD_UNCHANGED, apng_frames));
+    EXPECT_EQ(expected_frame_count, apng_frames.size());
+
+    apng_frames.clear();
+    // Test saving the animation frames as individual still images.
+    EXPECT_TRUE(imwrite(output, s_animation.frames));
+
+    // Read back the still images into a vector of Mats.
+    EXPECT_TRUE(imreadmulti(output, apng_frames));
+
+    // Expect all frames written as multi-page image
+    EXPECT_EQ(expected_frame_count, apng_frames.size());
```

Comment:
how about also testing the frames are the same as the input ones ?

Reply:
CV_16U Mats converted to CV_8U internally. so i did not do frames comparison. frankly handling CV_16U probably crude and need extra work if needed. i did not find 16U APNGs to test well.